estimagic/optimization/nlopt_optimizers.py

```diff
@@ -37,19 +44,25 @@ def nlopt_bobyqa(
 
     The implementation is derived from the BOBYQA subroutine of M. J. D. Powell.
 
-    ADD REMAINING EXPLANATIONS FROM NLOPT DOCUMENTATION HERE.
+    The algorithm performs derivative free bound-constrained optimization using
+    an iteratively constructed quadratic approximation for the objective function.
+    Due to its use of quadratic appoximation, the algorithm may perform poorly
+    for objective functions that are not twice-differentiable.
+
+    For details see:
+    M. J. D. Powell, "The BOBYQA algorithm for bound constrained optimization
+    without derivatives," Department of Applied Mathematics and Theoretical
+    Physics, Cambridge England, technical report NA2009/06 (2009).
 
     ``nlopt_bobyqa`` supports the following ``algo_options``:
 
     - convergence.relative_params_tolerance (float):  Stop when the relative movement
       between parameter vectors is smaller than this.
     - convergence.relative_criterion_tolerance (float): Stop when the relative
       improvement between two iterations is smaller than this.
-    - stopping.max_criterion_evaluations (int): If the maximum number of function
+    - stopping_max_criterion_evaluations (int): If the maximum number of function
```

Comment:
We actually want to use a `.` instead of `_` in the docstring here. In general we use dots after the prefixes `stopping`, `convergence`, `trustregion`, `genetic` and `linesearch`. It does not matter that the arguments themselves have an underscore. We replace the dot internally, but we want it because it makes the grouping of algo options more visible. 

So please use dots after the above mentioned prefixes also in all the other docstrings.

Reply:
Postponed to #234